transport/api/src/lib.rs

```diff
@@ -942,14 +942,22 @@ where
                 warn!(peer = peer.to_string(), "Manual ping to peer timed out");
                 return Err(ProtocolError::Timeout.into());
             }
-            Either::Right(_) => info!("Manual ping succeeded"),
+            Either::Right((Ok(_), _)) => info!("Manual ping succeeded"),
+            Either::Right((Err(e), _)) => {
+                // Ping errors need to be forwarded here
+                error!("Manual ping failed: {e}");
+                return Err(e.into());
+            }
         };
 
-        Ok(self
-            .network
-            .get(peer)
-            .await?
-            .map(|status| status.last_seen.as_unix_timestamp().saturating_sub(start)))
+        let peer_status = self.network.get(peer).await?.ok_or(HoprTransportError::NetworkError(
+            errors::NetworkingError::NonExistingPeer,
+        ))?;
```

Comment:
Is this true though? Because if you never contacted a peer, there's no last seen record. And if there's no record and you need to record a failure, this might not work.

Reply:
I believe this might an impossible error, due to 
````
if let Err(e) = self.network.add(peer, PeerOrigin::ManualPing, vec![]).await {
            error!("Failed to store the peer observation: {e}");
        }
````
on L934. Am I right ?